extensions/amp-date-picker/0.1/dates-list.js

```diff
@@ -80,7 +80,7 @@ export class DatesList {
       }
     }
     const rruleDates = this.rrulestrs_
-      .map(rrule => rrule.after(date))
+      .map((/** {RRule} */ rrule) => rrule.after(date))
```

Comment:
I think these would be eliminated if we properly typed `tryParseRrulestr` below.

Reply:
Don't think it's possible until https://github.com/google/closure-compiler/issues/3041 is fixed.

```js
extensions/amp-date-picker/0.1/dates-list.js:83: ERROR - Possible violation: ChildNode.p.after() is unusual. Please use Node.p.insertBefore()
The type information available for this expression is too loose to ensure conformance.
      .map(rrule => rrule.after(date))
                    ^^^^^^^^^^^

extensions/amp-date-picker/0.1/dates-list.js:113: ERROR - Possible violation: ChildNode.p.before() is unusual. Please use Node.p.insertBefore()
The type information available for this expression is too loose to ensure conformance.
      const rruleDay = this.moment_(rrule.before(nextDate));
                                    ^^^^^^^^^^^^

extensions/amp-date-picker/0.1/dates-list.js:141: ERROR - Bad type annotation. Unknown type RRule
 * @return {?RRule}
             ^
```

Reply:
Just import `RRule`? We're using the code anyways, having a pure type import isn't saving us any bytes.

Reply:
Importing `RRule` doesn't work trivially with the code as it's written (see errors above). I'll let you tackle this in a separate PR if you find an elegant way around it :)